server.go

```diff
@@ -1275,6 +1305,11 @@ func (s *server) addPeer(p *peer) {
 	// channel router so we can synchronize our view of the channel graph
 	// with this new peer.
 	go s.authGossiper.SynchronizeNode(p.addr.IdentityKey)
+
+	// Check if there are listeners waiting for this peer to come online.
+	for _, con := range s.peerConnectedListeners[pubStr] {
+		close(con)
```

Comment:
I believe this may return an `nil` value if the key has not yet been populated. As a result, this may result in a panic due to attempting to close a `nil` channel.

Reply:
If not populated, a nil value will be returned from the map, but I think ranging over nil slices is safe. I created an example here:
https://play.golang.org/p/S1Rhxjk2Ha

Reply:
whoa, you are totally right! I pushed a fix, PTAL. Good catch!

Reply:
LGTM 😎